src/gazebo_irlock_plugin.cpp

```diff
@@ -80,27 +83,64 @@ void IRLockPlugin::Load(sensors::SensorPtr _sensor, sdf::ElementPtr _sdf)
 
   this->camera->SetActive(true);
 
+  // Get the root model name
+  vector<std::string> names_splitted;
+  boost::split(names_splitted, scopedName, boost::is_any_of("::"));
+  names_splitted.erase(std::remove_if(begin(names_splitted), end(names_splitted),
+                            [](const string& name)
+                            { return name.size() == 0; }), end(names_splitted));
+
+  // get the root model name
+  const string rootModelName = names_splitted.front();
+
+  // store the model name
+  model_name_ = names_splitted.at(0);
+
+  std::cout << "[Gazebo Irlock Plugin] Irlock camera attached to: "<< model_name_ << std::endl; 
 }
 
 void IRLockPlugin::OnUpdated()
 {
+
+  // Store the pointer to the model.
+  if (model_ == NULL)
+#if GAZEBO_MAJOR_VERSION >= 9
+    model_ = this->world->ModelByName(model_name_);
+#else
+    model_ = this->world->GetModel(model_name_);
+#endif
+
   // Get the current simulation time.
 #if GAZEBO_MAJOR_VERSION >= 9
   common::Time now = world->SimTime();
+  ignition::math::Pose3d T_W_I = model_->WorldPose();
 #else
   common::Time now = world->GetSimTime();
+  ignition::math::Pose3d T_W_I = ignitionFromGazeboMath(model_->GetWorldPose());
 #endif
 
+  /* Get the attitude of the drone to convert observation from body to vehicle-carried NED frame */
+  ignition::math::Quaterniond& att_W_I = T_W_I.Rot();
+
   gazebo::msgs::LogicalCameraImage img = this->camera->Image();
 
   for (int idx = 0; idx < img.model_size(); idx++) {
 
     gazebo::msgs::LogicalCameraImage_Model model = img.model(idx);
 
-    if (model.has_name() && model.name() == "irlock_beacon") {
+    if (model.has_name() && model.name() == "land_pad") {
```

Comment:
Maybe better to keep the model name as an sdf parameter? Why do we need to check anyway? We already know which model this plugin is part of

Reply:
From my understanding, the logical camera reports every model in its field of view `gazebo::msgs::LogicalCameraImage_Model model = img.model(idx);`. However, we only want to use the relative pose information if the model is a target.

Reply:
Okay, could we then at least expose it as a sdf parameter?

Reply:
Yes, done in https://github.com/PX4/PX4-SITL_gazebo-classic/pull/950/commits/85390e89165d40b3cb1894bee218ec5fc1277fa5